### PR TITLE
Feature/etm keyedvectors embeddings input file

### LINKED ISSUE #32: ETM - Possibility of using KeyedVectors input for pre-trained W2V embeddings
* OCTIS version: 1.9.0
* Python version: 3.7.6
* Operating System: Ubuntu 20.04 LTS

### Description

Hi, this is more of a question than anything else. I've seen that for ETM model training, we must pass an embeddings path corresponding to a "pickled" file. However, I need to execute ETM with rather large embeddings. There's any intent on implementing a [```gensim.models.KeyedVectors```](https://radimrehurek.com/gensim/models/keyedvectors.html) based (or something like that) embeddings input for this model? I've implemented something like that for [an etm package of mine](https://github.com/lffloyd/embedded-topic-model/blob/bea451a28f32cdfcb47bc5fabd9efd9a0ecba2c8/embedded_topic_model/models/etm.py#L161), but yours' has all I need to execute model optimization. Would a PR on this matter be accepted?

Anyway, cheers for the nice work, this package is really great!

### What I Did

Gave a look at [here](https://github.com/MIND-Lab/OCTIS/blob/69988cae2c9e9b0db75a28cda4784f0101069cc5/octis/models/ETM.py#L345).

### PR DESCRIPTION
**Pull request by lffloyd**
Resolves #32, #37.

Adds support for original-formatted word2vec and [```gensim.models.KeyedVectors```](https://radimrehurek.com/gensim/models/keyedvectors.html)-formatted embeddings files for ETM model training.
Updates some details on the ```CONTRIBUTING``` section of the documentation.

### CODE REVIEW

File: octis/models/ETM.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ class ETM(BaseETM):
     def __init__(self, num_topics=10, num_epochs=100, t_hidden_size=800, rho_size=300, embedding_size=300,
                  activation='relu', dropout=0.5, lr=0.005, optimizer='adam', batch_size=128, clip=0.0,
                  wdecay=1.2e-6, bow_norm=1, device='cpu', top_word=10, train_embeddings=True, embeddings_path=None,
-                 use_partitions=True):
+                 embeddings_type='pickle', binary_embeddings=True, use_partitions=True):
```
Comment (silviatti) [inline octis/models/ETM.py:18]: Can you add the comments to explain the parameters "embeddings_type" and "binary_embeddings"? Including the type and default values (I'll take care of the others parameters).

File: octis/models/base_etm.py [hunk 2]
```diff
@@ -45,24 +46,35 @@ def preprocess(vocab2id, train_corpus, test_corpus=None, validation_corpus=None)
...
+                                        self.hyperparameters['binary_embeddings'])
+        embeddings = np.zeros((len(self.vocab.keys()), self.hyperparameters['embedding_size']))
+        for i, word in enumerate(self.vocab.values()):
+            try:
+                embeddings[i] = vectors[word]
+            except KeyError:
+                embeddings[i] = np.random.normal(scale=0.6, size=(self.hyperparameters['embedding_size'],))
+        self.embeddings = torch.from_numpy(embeddings).to(self.device)
+
+    def _load_word_vectors(self, embeddings_path, embeddings_type, binary_embeddings=True):
```
Comment (silviatti) [inline octis/models/base_etm.py:62]: Can you add the comments to explain the parameters of the method? Including the type and default values

**Review by silviatti — CHANGES_REQUESTED** [review]
Thank you for your work! :) I just added a few comments to address before merging

**Comment by lffloyd** [comment]
> Thank you for your work! :) I just added a few comments to address before merging

Thanks for the feedback, @silviatti! I've added docstrings for the changed methods, at your request.

File: octis/models/ETM.py (hunk 1 above)
Comment (lffloyd) [reply octis/models/ETM.py:18]: Done

File: octis/models/base_etm.py (hunk 2 above)
Comment (lffloyd) [reply octis/models/base_etm.py:62]: Done

**Comment by lffloyd** [comment]
Hi @silviatti . Updated the PR with [a commit](https://github.com/MIND-Lab/OCTIS/pull/36/commits/086d7ebe6aace2908295b1488580b5e47809f141) adding support for headerless original word2vec embeddings file in textual format. This feature provides support for the Dieng's pretrained embeddings found [here](https://bitbucket.org/diengadji/embeddings/src/master/), as requested by [cayaluke's comment](https://github.com/MIND-Lab/OCTIS/issues/32#issuecomment-963716020) on #32. Previously, my PR supported only regular files including the header.

**Review by silviatti — APPROVED** [review]
Thank you again for your work and patience :) I'm going to merge the changes and I'll soon make the new release.

Silvia

**Pull request merged**